model/QtiTestDeletedListener.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2023 (original work) Open Assessment Technologies SA.
+ */
+
+namespace oat\taoMediaManager\model;
+
+use oat\generis\model\data\Ontology;
+use oat\tao\model\media\TaoMediaException;
+use oat\tao\model\media\TaoMediaResolver;
+use oat\tao\model\resources\Exception\ClassDeletionException;
+use oat\tao\model\resources\Exception\PartialClassDeletionException;
+use oat\tao\model\resources\Service\ClassDeleter;
+use oat\taoMediaManager\model\Specification\MediaClassSpecification;
+use oat\taoQtiTest\models\event\QtiTestDeletedEvent;
+use core_kernel_classes_Class;
+use core_kernel_classes_Resource;
+use tao_helpers_Uri;
+use Psr\Log\LoggerInterface;
+use Exception;
+
+class QtiTestDeletedListener
+{
+    private LoggerInterface $logger;
+    private Ontology $ontology;
+    private ClassDeleter $classDeleter;
+    private TaoMediaResolver $taoMediaResolver;
+    private MediaClassSpecification $mediaClassSpecification;
+    private MediaService $mediaService;
+
+    public function __construct(
+        LoggerInterface $logger,
+        MediaService $mediaService,
+        MediaClassSpecification $mediaClassSpecification,
+        Ontology $ontology,
+        ClassDeleter $classDeleter,
+        TaoMediaResolver $taoMediaResolver
+    ) {
+        $this->logger = $logger;
+        $this->mediaService = $mediaService;
+        $this->mediaClassSpecification = $mediaClassSpecification;
+        $this->ontology = $ontology;
+        $this->classDeleter = $classDeleter;
+        $this->taoMediaResolver = $taoMediaResolver;
+    }
+
+    /**
+     * @throws PartialClassDeletionException
+     * @throws ClassDeletionException
+     * @throws Exception
+     */
+    public function handleQtiTestDeletedEvent(QtiTestDeletedEvent $event): void
+    {
+        $assetIds = [];
+
+        foreach (array_unique($event->getReferencedResources()) as $ref) {
+            try {
+                $asset = $this->taoMediaResolver->resolve($ref);
+                $assetIds[] = $asset->getMediaIdentifier();
```

Comment:
Variable is not used

```suggestion
                $assetIds[] = $this->taoMediaResolver->resolve($ref)->getMediaIdentifier();
```

Reply:
Changed: b47f860